### PR TITLE
feat(video-conferencing-service): SFO-63 adds videochat session related APIs

### PR DESCRIPTION
**Pull request by vineet-suri**
## Description

Adds videochat session APIs

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/video-conferencing-service/migrations/20200526081137-db-init.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var dbm;
```
Comment (bhavyasf) [inline services/video-conferencing-service/migrations/20200526081137-db-init.js:3]: use `const` or `let` everywhere.

File: services/video-conferencing-service/src/__tests__/unit/controllers/video-chat.controller.unit.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        sessionOptions,
+        meetingLink,
+      );
+      expect(result.sessionId).to.be.a.String();
+      expect(result.token).to.be.a.String();
+      sinon.assert.called(find);
+    });
+
+    it('gives an error for invalid meeting link', async () => {
+      // const sessionOptions = getSessionOptions({meetingLink: ''});
```
Comment (bhavyasf) [inline services/video-conferencing-service/src/__tests__/unit/controllers/video-chat.controller.unit.ts:105]: why this is commented?

File: services/video-conferencing-service/src/providers/vonage/types.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+ *   @param stylesheet if you specify a "custom" @param type, set the stylesheet property of the layout object to the stylesheet.
+ * @param name (String) — (Optional) The name of the archive (for your own identification)
+ * @param outputMode String) — (Optional) Whether all streams in the archive are recorded to a single file
+ * ("composed", the default) or to individual files ("individual")
+ * @param resolution  (String) — (Optional) The resolution of the archive, either "640x480" (SD, the default)
+ * or "1280x720" (HD). This property only applies to composed archives. If you set this property and set the
+ * outputMode property to "individual", the call to the REST method results in an error.
+ */
+export interface VonageArchive {
+  sessionId: string;
```
Comment (bhavyasf) [inline services/video-conferencing-service/src/providers/vonage/types.ts:94]: this is removed in my PR.
No need for VonageArchive Interface. Already defined in Opentok

File: services/video-conferencing-service/migrations/20200526081137-db-init.js (hunk 1 above)
Comment (vineet-suri) [reply services/video-conferencing-service/migrations/20200526081137-db-init.js:3]: this is auto-generated file by db-migrate package

File: services/video-conferencing-service/src/__tests__/unit/controllers/video-chat.controller.unit.ts (hunk 2 above)
Comment (vineet-suri) [reply services/video-conferencing-service/src/__tests__/unit/controllers/video-chat.controller.unit.ts:105]: Anit is working on this

File: services/video-conferencing-service/src/providers/vonage/types.ts (hunk 3 above)
Comment (vineet-suri) [reply services/video-conferencing-service/src/providers/vonage/types.ts:94]: Will rebase when the Archiving code is merged.

**Comment by samarpan-b** [comment]
Feedback is same as pull request #3 . Fix those first and get it merged. We will then rebase and merge this 1.

**Pull request closed without merging**